katas/args/rubiana/args.test.js

```diff
@@ -0,0 +1,55 @@
+    expect(schemaClassInstance.splitInputStr('-l -p 8080 -d /usr/logs')).toEqual(['-l', '-p', '8080', '-d', '/usr/logs']);
+});
+
+test('isFlag() returns true if is a flag or false if is a negative number.', () => {
+    expect(schemaClassInstance.isFlag('-p')).toBe(true);
+    expect(schemaClassInstance.isFlag('-2')).toBe(false);
+});
+
+test('validFlag() check if the flag exists in our schema', () => {
+    expect(schemaClassInstance.validFlag('p')).toBe(true);
+    expect(schemaClassInstance.validFlag('8')).toBe('Invalid! The Flag does not exist.');
+    expect(schemaClassInstance.validFlag('z')).toBe('Invalid! The Flag does not exist.');
+});
+
+test('defaultValue() returns the default value of a flag', () => {
+    expect(schemaClassInstance.defaultValue('l')).toBe(false);
+    expect(schemaClassInstance.defaultValue('p')).toBe(0);
+    expect(schemaClassInstance.defaultValue('d')).toBe('');
+});
+
+test('flagIndex() returns the index of the flag in our schema', () => {
+    expect(schemaClassInstance.flagIndex('l')).toBe(0);
+    expect(schemaClassInstance.flagIndex('p')).toBe(1);
+    expect(schemaClassInstance.flagIndex('d')).toBe(2);
+});
+
+test('setNewFlag() returns a message if the flag was successfully added to the schema or it was missing a parameter', () => {
+    expect(schemaClassInstance.setNewFlag('r', 'rubiana', String, 'student')).toBe('New Flag "r" added to the Schema');
+    expect(schemaClassInstance.setNewFlag('r', String, 'student')).toBe('Missing parameters.');
+});
+
+test('checkInput() returns true if all the inputs in the object is complete, or false if one of then if undefined', () => {
+    expect(schemaClassInstance.checkInput({ flag: 'r', name: 'rubiana', type: String, defaultValue: 'student' })).toBe(true);
+    expect(schemaClassInstance.checkInput({ flag: 'r', name: 'rubiana', type: String, defaultValue: undefined })).toBe(false);
+});
+
+test('displaySchema() console.log() the current schema', () => {
+    expect(schemaClassInstance.displaySchema()).toEqual(console.log([{ flag: 'l', name: 'logging', type: Boolean, defaultValue: false }, { flag: 'p', name: 'port', type: Number, defaultValue: 0, }, { flag: 'd', name: 'directory', type: String, defaultValue: '' }]));
+    schemaClassInstance.setNewFlag('r', 'rubiana', String, 'student');
+    expect(schemaClassInstance.displaySchema()).toEqual(console.log([{ flag: 'l', name: 'logging', type: Boolean, defaultValue: false }, { flag: 'p', name: 'port', type: Number, defaultValue: 0, }, { flag: 'd', name: 'directory', type: String, defaultValue: '' }, { flag: 'r', name: 'rubiana', type: String, defaultValue: 'student' }]));
+});
+
+const parserClassInstance = new parserClass;
+test('output() supposed to bring the flags and the correspondents values or the default values when missing', () => {
+    expect(parserClassInstance.output('-l -p 8080 -d /usr/logs')).toBe(console.log('Flag: -l Value: false Flag: -p Value: 8080 Flag: -d Value: /usr/logs'));
```

Comment:
Maybe the result of `output()` function should be some sort of object instead of just an string